tests

```diff
@@ -294,6 +306,141 @@ EOF
 	./safe export secret/import >t/home/got 2>t/home/errors
 	yamlok
 
+	testing ${version} x509 commands
+	./safe x509 issue                                                ; exitok $? 1 # need a path
+	./safe x509 issue --signed-by secret/enoent                      ; exitok $? 1 # still need a name...
+	./safe x509 issue --signed-by secret/enoent secret/cert          ; exitok $? 1 # ca must exist
+
+	./safe x509 issue secret/x509/self-signed --name www.example.com ; exitok $? 0
+	./safe x509 validate secret/x509/self-signed                     ; exitok $? 0
+	./safe exists secret/x509/self-signed:certificate                ; exitok $? 0
+	./safe exists secret/x509/self-signed:key                        ; exitok $? 0
+	./safe exists secret/x509/self-signed:combined                   ; exitok $? 0
+	./safe exists secret/x509/self-signed:serial                     ; exitok $? 1  # not a CA
+	./safe exists secret/x509/self-signed:crl                        ; exitok $? 1  # not a CA
+	./safe x509 check secret/x509/self-signed                        ; exitok $? 0
+	./safe x509 check secret/x509/self-signed --for www.example.com  ; exitok $? 0
+	./safe x509 check secret/x509/self-signed --for mail.example.com ; exitok $? 1
+	./safe x509 check secret/x509/self-signed --for test@example.com ; exitok $? 1
+	./safe x509 check secret/x509/self-signed --for 10.40.5.6        ; exitok $? 1
+
+	./safe x509 issue secret/fail-without-name                       ; exitok $? 1
+	./safe exists secret/fail-without-name:certificate               ; exitok $? 1
+	./safe exists secret/fail-without-name:key                       ; exitok $? 1
+	./safe exists secret/fail-without-name:combined                  ; exitok $? 1
+	./safe exists secret/fail-without-name:serial                    ; exitok $? 1
+	./safe exists secret/fail-without-name:crl                       ; exitok $? 1
```

Comment:
I'm more inclined to go with (a).  If you `safe ssh` over the top of an existing path, safe only overrides the keys related to SSH.  If someone _wants_ to co-locate other credentials with their certificate / CA, that's their business and I assume they have a compelling reason.

That being said, accidentally overwriting a CA with a certificate will cause problems because in some cases, we check the _existence_ of `crl` and `serial` to determine if we should do CA-like things.  Instead, we should always consult the BasicConstraints / IsCA:True of the certificate itself.  I will fix that, add some more tests and push an addendum to this PR.